libraries/CMakeLists.txt

```diff
@@ -41,6 +41,7 @@ set( LIBRARIES
    lib-numeric-formats
    lib-effects
    lib-time-frequency-selection
+   lib-snapping
```

Comment:
I didn't see yet what happens to src/Snap.*

In a branch that I haven't shared yet, I made it the home of the new attached object, I moved the functions that draw yellow lines out into TrackArt.cpp, and I moved them to lib-time-frequency-selection

I also found a way to break dependency of ProjectSelectionManager on WaveTrack, and then I moved that too into lib-time-frequency-selection